storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -219,17 +219,44 @@ public void setBlobMeta(String key, SettableBlobMeta meta, Subject who) throws A
     @Override
     public void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
         validateKey(key);
-        checkForBlobOrDownload(key);
-        SettableBlobMeta meta = getStoredBlobMeta(key);
-        _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+
+        boolean checkPermission = true;
+        try {
+            checkForBlobOrDownload(key);
+        } catch (KeyNotFoundException e) {
+            LOG.warn("Detected deletion from ZK for the key {} while pulling the latest change. Needs to be deleted. key: {}", key);
+            checkPermission = false;
+        }
+
+        if (checkPermission) {
```

Comment:
The meaning for KeyNotFoundException is different between `checkForBlobOrDownload` and `getStoredBlobMeta`. First one indicates ZK node is not exist and second one indicates META blob file in local is not exist.